### PR TITLE
Use Go cipher suites to find TLS suite string name

### LINKED ISSUE #5075: tls: clarify/cleanup TLS cipher list
We define a [map](https://github.com/grpc/grpc-go/blob/4d58dd98db75e180ebf67cce127db5068ff7ab3e/credentials/tls.go#L209) from TLS cipher suite consts to string which is used [here](https://github.com/grpc/grpc-go/blob/4d58dd98db75e180ebf67cce127db5068ff7ab3e/credentials/tls.go#L50)

In the Go stdlib some of these are marked as insecure. See [here](https://github.com/golang/go/blob/2580d0e08d5e9f979b943758d3c49877fb2324cb/src/crypto/tls/cipher_suites.go#L81)

Some questions that we should try answering:
- Do we need to keep this map in sync with the latest Go version?
- Go stdlib returns a list of [secure cipher suites](https://github.com/golang/go/blob/2580d0e08d5e9f979b943758d3c49877fb2324cb/src/crypto/tls/cipher_suites.go#L52) and [insecure cipher suites](https://github.com/golang/go/blob/2580d0e08d5e9f979b943758d3c49877fb2324cb/src/crypto/tls/cipher_suites.go#L81). Should we use these and dynamically populate our `cipherSuites` [map](https://github.com/grpc/grpc-go/blob/4d58dd98db75e180ebf67cce127db5068ff7ab3e/credentials/tls.go#L209)?

The way we use our map is not very problematic in the sense that if we don't find the actual cipher suite being used in our map, we end up setting the corresponding value returned to channelz to an empty string. So, this won't cause a panic or cause us to use a cipher suite which is not supported. But it would be worth thinking about this and figuring if we can do anything better here.

### PR DESCRIPTION
**Pull request by bcr**
Replaces the map of cipher suite IDs with lookups in the globally known cipher suites. Fixes #5075 

RELEASE NOTES: none

### CODE REVIEW

File: credentials/tls.go [hunk 1]
```diff
@@ -44,10 +44,27 @@ func (t TLSInfo) AuthType() string {
 	return "tls"
 }
 
+// cipherSuiteLookup returns the string version of a TLS cipher suite ID.
+func cipherSuiteLookup(cipherSuiteID uint16) string {
+	suites := tls.CipherSuites()
+	insecureSuites := tls.InsecureCipherSuites()
+	for i := range suites {
```
Comment (dfawley) [inline credentials/tls.go:51]: Please remove the local var and use the value part of the range iterator:

```go
for _, s := range tls.CipherSuites() {
	if s.ID == cipherSuiteID {
		return s.Name
	}
}
<etc>
```

File: credentials/tls.go [hunk 2]
```diff
@@ -44,10 +44,27 @@ func (t TLSInfo) AuthType() string {
...
+		if suites[i].ID == cipherSuiteID {
+			return suites[i].Name
+		}
+	}
+	for i := range insecureSuites {
+		if insecureSuites[i].ID == cipherSuiteID {
+			return suites[i].Name
+		}
+	}
+	return ""
```
Comment (dfawley) [inline credentials/tls.go:61]: Let's do something like this instead, which will be a little nicer if it ever happens (though it should really be impossible): `return fmt.Sprintf("unknown ID: %v", cipherSuiteID)`

**Review by dfawley — CHANGES_REQUESTED** [review]

**Comment by dfawley** [comment]
Thanks for the PR!  LGTM except for a couple minor tweaks.

File: credentials/tls.go (hunk 1 above)
Comment (bcr) [reply credentials/tls.go:51]: This was actually my original approach, but I worried that I was doing a struct copy on each iteration. I'll make the change.

Comment (dfawley) [reply credentials/tls.go:51]: That should be okay..  If we want to optimize this, we should just build the map we used to have using these functions.  (I'm OK either way.)

**Review by dfawley — APPROVED** [review]

**Review by arvindbr8 — APPROVED** [review]
LGTM

**Comment by bcr** [comment]
Thanks folks -- I'm good on my end too.

**Pull request merged**